File: mobile/src/main/java/treehou/se/habit/core/db/model/controller/CellRowDB.java
package treehou.se.habit.core.db.model.controller;

import io.realm.Realm;
import io.realm.RealmList;
import io.realm.RealmObject;

public class CellRowDB extends RealmObject {

    private ControllerDB controller;
    private RealmList<CellDB> cells = new RealmList<>();

    public RealmList<CellDB> getCells() {
        return cells;
    }

    public void setCells(RealmList<CellDB> cells) {
        this.cells = cells;
    }

    public ControllerDB getController() {
        return controller;
    }

    public void setController(ControllerDB controller) {
        this.controller = controller;
    }

    public CellDB addCell(Realm realm){
        CellDB cell = new CellDB();
        cell.setCellRow(this);
        CellDB cellDB = CellDB.save(realm, cell);

        realm.beginTransaction();
        cells.add(cellDB);
        realm.commitTransaction();

        return cell;
    }
}

File: mobile/src/main/java/treehou/se/habit/core/db/model/controller/CellRowDB.kt
package treehou.se.habit.core.db.model.controller

import io.realm.Realm
import io.realm.RealmList
import io.realm.RealmObject

open class CellRowDB : RealmObject() {

    var controller: ControllerDB? = null
    var cells = RealmList<CellDB>()

    fun addCell(realm: Realm): CellDB {
        val cell = CellDB()
        cell.cellRow = this
        val cellDB = CellDB.save(realm, cell)

        realm.beginTransaction()
        cells.add(cellDB)
        realm.commitTransaction()

        return cell
    }
}
